testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml

```
@@ -39,12 +39,17 @@
     oc label csv -n "{{ app_ns }}" "{{ item.metadata.name }}" test-network-function.com/operator=target --overwrite
     oc annotate csv -n "{{ app_ns }}" "{{ item.metadata.name }}" test-network-function.com/operator_tests='["OPERATOR_STATUS"]' --overwrite
     
-    INSTALL_PLAN_NAME=$(oc get installplan -n "{{ app_ns }}" | grep "{{ item.metadata.name }}" | awk '{ print $1 }')
-    SUBSCRIPTION_NAMES=($(oc get installplan $INSTALL_PLAN_NAME -n "{{ app_ns }}" -o json | jq -r ".metadata.ownerReferences[].name"))
-    SUBSCRIPTION_NAMES_JSON=$(echo ${SUBSCRIPTION_NAMES[@]} | jq -R 'split (" ")')
-
-    oc annotate csv -n "{{ app_ns }}" "{{ item.metadata.name }}" test-network-function.com/subscription_name="$SUBSCRIPTION_NAMES_JSON" --overwrite
-
+    INSTALL_PLAN_NAMES=($(oc get installplan -n "{{ app_ns }}" | grep "{{ item.metadata.name }}" | awk '{ print $1 }'))
+    INSTALL_PLAN_NUMBER=$(echo -n ${INSTALL_PLAN_NAMES[@]} | wc -w)
+
+    if [ $INSTALL_PLAN_NUMBER -ne 0 ]; then
```

You can use bash built-in array length instead:
```suggestion
if [ ${#INSTALL_PLAN_NAMES[@]} -ne 0 ]; then
```

---

If you check the [commits](https://github.com/dci-labs/example-cnf-config/pull/39/commits/27d6e309f99de492e8e3b52fff5e615dd0e20483), I included this in the first version, but I don't know why, Ansible was compaining about this, so that's why I had to change it.

---

:thinking: This is odd, indeed.  I did a quick test, and seems to be working, I'm using a slightly different entry, but should be equivalent:

```
$ IP=($(oc get installplan -A | awk '/openshift/ {print $1}'))
echo ${#IP[@]}
3
```

Do you have a job where it failed,  we could figure out what's going on.

---

I figured this out, is due to jinja2 template the "way" to fix the issue with `{#` (jinja commenting) is to build char by char then let the template append it when rendered, but is too much hassle:

```yaml
- name: shell jinja
  hosts: localhost
  gather_facts: false
  tasks:
  - name: shell
    shell: |
      x=( uno dos tres )
      echo ${x}
      echo ${{ "{" }}{{ "#" }}x[@]}
```